plugins/urls/__init__.py

```diff
@@ -1,165 +1,228 @@
+            return self._disable_self()
+
+        if "catcher" in self.config:
+            self.logger.warn(
+                "========================================================="
+            )
+            self.logger.warn(
+                " It appears that you haven't updated your configuration."
+            )
+            self.logger.warn("")
+            self.logger.warn(
+                " Please check the example configuration and update your"
+            )
+            self.logger.warn(
+                " file to match."
+            )
+            self.logger.warn(
+                "========================================================="
+            )
+
+        self.channels = self.storage.get_file(
+            self, "data", Formats.YAML, "plugins/urls/channels.yml"
+        )
 
-        self.channels = self.storage.get_file(self, "data", YAML,
-                                              "plugins/urls/channels.yml")
         self.shortened = self.storage.get_file(
             self,
             "data",
-            DBAPI,
+            Formats.DBAPI,
             "sqlite3:data/plugins/urls/shortened.sqlite",
             "data/plugins/urls/shortened.sqlite",
             check_same_thread=False
         )
 
-        self.commands = CommandManager()
-        self.events = EventManager()
-
+        self.config.add_callback(self.reload)
         self.reload()
 
-        def message_event_filter(event=MessageReceived):
-            target = event.target
-            type_ = event.type
-
-            return type_ == "message" \
-                or isinstance(target, Channel) \
-                or isinstance(target, User)
-
-        self.add_shortener("tinyurl", self.tinyurl)
+        def message_event_filter(event):
+            return event.type == "message"
 
         self.events.add_callback("MessageReceived", self, self.message_handler,
                                  1, message_event_filter)
+
+        self.add_handler(WebsiteHandler(self), -100)
+        self.add_shortener(TinyURLShortener(self))
+
         self.commands.register_command("urls", self.urls_command, self,
                                        "urls.manage")
         self.commands.register_command("shorten", self.shorten_command, self,
                                        "urls.shorten", default=True)
 
-    def reload(self):
-        """
-        Reload files and create tables as necessary
-        """
+        if not self.factory_manager.running:
+            self.events.add_callback(
+                "ReactorStarted", self, self.send_event, 0
+            )
+        else:
+            self.send_event()
 
+    @property
+    def default_shortener(self):
+        shortener = self.config.get("default_shortener", "tinyurl")
+
+        return shortener if self.has_shortener(shortener) else "tinyurl"
+
+    def send_event(self, _=None):
+        self.events.run_callback("URLs/PluginLoaded", URLsPluginLoaded(self))
```

Comment:
There should really be a generic `PluginEnabled` event fired by the plugin manager (along with `PluginDisabled`, etc.). That's a separate issue, but it should really be a priority to replace this method before anything outside of this repo uses it.